services/api/pepc_boost_test.go

```diff
@@ -583,10 +591,32 @@ func TestNetworkIndependentCheckTxAndSenderValidity(t *testing.T) {
 				tracerError: "",
 				callTrace:   c.callTraces,
 			}
+			if c.tobSimulationError != "" {
+				backend.relay.blockSimRateLimiter = &MockBlockSimulationRateLimiter{
+					tobSimulationError: fmt.Errorf(c.tobSimulationError),
+				}
+			}
+
+			tobTxReqs := bellatrixUtil.ExecutionPayloadTransactions{Transactions: []bellatrix.Transaction{}}
+			for _, tx := range c.txs {
+				txByte, err := tx.MarshalBinary()
+				require.NoError(t, err)
+				tobTxReqs.Transactions = append(tobTxReqs.Transactions, txByte)
+			}
+			req := &common.TobTxsSubmitRequest{
+				ParentHash: parentHash,
+				TobTxs:     tobTxReqs,
+				Slot:       headSlot + 1,
+			}
+			jsonReq, err := req.MarshalJSON()
+			require.NoError(t, err)
+			rr := backend.requestBytes(http.MethodPost, tobTxSubmitPath, jsonReq, map[string]string{
+				"Content-Type": "application/json",
+			})
 
-			err := backend.relay.checkTxAndSenderValidity(c.txs, headSlot+1, common.TestLog)
 			if c.requiredError != "" {
-				require.Contains(t, err.Error(), c.requiredError)
+				require.Equal(t, http.StatusBadRequest, rr.Code)
+				require.Contains(t, rr.Body.String(), c.requiredError)
 			} else {
 				require.NoError(t, err)
```

Comment:
nit: return in the if and then the `require.NoError` can be in the outer scope (no else)